scripts/minikube.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash -e
+
+#https://github.com/rook/rook/blob/master/tests/scripts/minikube.sh
+
+function wait_for_ssh() {
+    local tries=100
+    while ((tries > 0)); do
+        if minikube ssh echo connected &>/dev/null; then
+            return 0
+        fi
+        tries=$((tries - 1))
+        sleep 0.1
+    done
+    echo ERROR: ssh did not come up >&2
+    exit 1
+}
+
+function copy_image_to_cluster() {
+    local build_image=$1
+    local final_image=$2
+    if [[ "${VM_DRIVER}" == "none" ]]; then
+        docker pull "${build_image}"
+        docker tag "${build_image}" "${final_image}"
+        return
+    fi
+    docker save "${build_image}" | (eval "$(minikube docker-env --shell bash)" && docker load && docker tag "${build_image}" "${final_image}")
+}
+
+# configure minikube
+MINIKUBE_VERSION=${MINIKUBE_VERSION:-"latest"}
+KUBE_VERSION=${KUBE_VERSION:-"v1.13.0"}
+MEMORY=${MEMORY:-"3000"}
+VM_DRIVER=${VM_DRIVER:-"virtualbox"}
+#configure image repo
+CEPHCSI_IMAGE_REPO=${CEPHCSI_IMAGE_REPO:-"quay.io/cephcsi"}
+K8S_IMAGE_REPO=${K8S_IMAGE_REPO:-"quay.io/k8scsi"}
+
+#feature-gates for kube
+K8S_FEATURE_GATES=${K8S_FEATURE_GATES:-"BlockVolume=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"}
+# install minikube
+function install_minikube() {
+    if type minikube >/dev/null 2>&1; then
+        local version
+        version="$(minikube version)"
+        echo "minikube already installed with ${version}"
+        return
+    fi
+
+    echo "Installing minikube. Version: ${MINIKUBE_VERSION}"
+    curl -Lo minikube https://storage.googleapis.com/minikube/releases/"${MINIKUBE_VERSION}"/minikube-linux-amd64 && chmod +x minikube && mv minikube /usr/local/bin/
+}
+
+function install_kubectl() {
+    # Download kubectl, which is a requirement for using minikube.
+    echo "Installing kubectl. Version: ${KUBE_VERSION}"
+    curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/"${KUBE_VERSION}"/bin/linux/amd64/kubectl && chmod +x kubectl && mv kubectl /usr/local/bin/
+}
+
+case "${1:-}" in
+up)
+    install_minikube
+    #if driver  is none install kubectl with KUBE_VERSION
+    if [[ "${VM_DRIVER}" == "none" ]]; then
+        mkdir -p "$HOME"/.kube "$HOME"/.minikube
+        install_kubectl
+    fi
+
+    echo "starting minikube with kubeadm bootstrapper"
+    minikube start --memory="${MEMORY}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --vm-driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}"
+
+    DISK="sda1"
```

Comment:
(suggestion) Do we want to retain this block as in the original script (although this is correct) for easy merges in the future?

Reply:
I prefer doing this way, this will help us to get the driver used to deploy Kube (currently Virtualbox is the default in minikube, later if they change the default to something else we don't need to change any part of the code here).